test/create_game_section/Lobby/lobby_widget_test.dart

```diff
@@ -68,18 +74,16 @@ void main() {
         MockDatabaseService(controller: createdGamesStreamContoller);
     final mockUserData = MockUser().userData;
 
-    //sign in the user
-    controller.add(mockUserData);
     // create the widget under test
-    await tester.pumpWidget(
-      MyApp(
-        mockAuthService,
-        navigation.navigatorKey,
-        mockDBService,
-        navigation,
-      ),
-    );
+    await tester.pumpWidget(MyApp(
+      authService: mockAuthService,
+      navigatorKey: navigation.navigatorKey,
+      databaseService: mockDBService,
+      navigation: navigation,
+    ));
 
+    //sign in the user
+    controller.add(mockUserData);
```

Comment:
we were adding stream data before pumping test widget, now because we initialise services in myApp, not in main(), we can only add stream data after pumping the testwidget. (services are only available after widget is built)

Reply:
can ignore this @nickmeinhold , was more me talking out loud to understand it lol

Reply:
Got it! 👍